tests/erc20/test_erc20_mintable_burnable.py

```diff
@@ -299,3 +312,36 @@ def test_raw_logs(c, w3, get_log_args):
 
 def test_failed_send_in_withdraw(c, w3):
     pass
+
+
+def test_bad_transfer(c_bad, w3, assert_tx_failed):
+    # Ensure transfer fails if it would otherwise overflow balance when totalSupply is corrupted
+    minter, a1, a2 = w3.eth.accounts[0:3]
+    c_bad.mint(a1, MAX_UINT256, transact={'from': minter})
+    c_bad.mint(a2, 1, transact={'from': minter})
```

Comment:
(Just to confirm) This line checks that total_supply doesn't increase?

Reply:
The purpose of `bad_code` is to check the `balances` storage variable correctly detects overflow and reverts.
In `transfer`,  `transferFrom` and `burn`, `totalSupply` is increased/decreased before `balances` so we can't check that in the correct code. That's why we remove these codes as `bad_code` .

Reply:
got it!